docs/dev/design/new-notifications-system.rst

```diff
@@ -0,0 +1,459 @@
+Notification system: a new approach after lot of discussions
+============================================================
+
+Notifications have been a recurrent topic in the last years.
+We have talked about different problems and solution's approaches during these years.
+However, due to the complexity of the change, and without having a clear path, it has been hard to prioritize.
+
+We've written a lot about the problems and potential solutions for the current notification system.
+This is a non-complete list of them just for reference:
+
+* https://github.com/readthedocs/readthedocs.org/issues/4226
+* https://github.com/readthedocs/readthedocs.org/issues/3399
+* https://github.com/readthedocs/meta/discussions/126
+* https://github.com/readthedocs/readthedocs.org/issues/5909
+* https://github.com/readthedocs/readthedocs.org/issues/9110
+* https://github.com/readthedocs/readthedocs.org/issues/9279
+* https://github.com/readthedocs/readthedocs.org/issues/4235
+
+At the offsite in Portland, Anthony and myself were able to talk deeply about this and wrote a bunch of thoughts in a Google Doc.
+We had pretty similar ideas and we thought we were solving most of the problems we identified already.
+
+I read all of these issues and all the discussions I found and wrote this document that summarizes my proposal:
+*create a new notification system that we can customize and expand as we need in the future*:
+
+* A Django model to store the notifications' data
+* API endpoints to retrieve the notifications for a particular resource (User, Build, Project, Organization)
+* Frontend code to display them (*outside the scope of this document*)
+
+
+Goals
+-----
+
+* Keep raising exceptions for errors from the build process
+* Ability to add non-error notifications from the build process
+* Add extra metadata associated to the notification: icon, color, header, body, etc
+* Support different types of notifications (e.g. error, warning, note, tip)
+* Re-use the new notification system for product updates (e.g. new features, deprecated config keys)
+* Message content lives on Python classes that can be translated and formatted with objects (e.g. Build, Project)
+* Message could have richer content (e.g. HTML code) to generate links and emphasis
+* Notifications have trackable state (e.g. unread (default)=never shown, read=shown, dismissed=don't show again)
+* An object (e.g. Build, Organization) can have more than 1 notification attached
+* Remove hardcoded notifications from the templates
+* Notifications can be attached to Project, Organization, Build and User models
+* Specific notifications can be shown under the user's bell icon
```

Comment:
> We should find a better way of describing this

Do you mean regarding "naming"? Like, not calling it `bell`? If so, we can use something more direct, like `user`, since it's a direct user-specific notification.

Reply:
I updated this to `news` in 404615ab6